website_sale_product_contract_gift/views/templates.xml

```diff
@@ -0,0 +1,118 @@
+                <h3>This order is a gift</h3>
+                <p>
+                    <b>Date of the gift:</b> <span
+                        t-field="website_sale_order.gift_date"
+                    />
+                </p>
+            </div>
+        </xpath>
+    </template>
+
+    <template id="checkout" inherit_id="website_sale.checkout">
+        <!-- Add date for gift formular -->
+        <xpath
+            expr="//div[hasclass('oe_cart')]//div[hasclass('row')]"
+            position="before"
+        >
+            <div class="row" id="date_for_gift" t-if="website_sale_order.is_gift">
+                <div class="col-lg-12">
+                    <h3
+                        class="o_page_header mt8"
+                    >When should the gift be delivered ?</h3>
+                </div>
+                <div class="col-lg-12">
+                    <form
+                        action="/shop/checkout"
+                        method="post"
+                        name="date_for_gift"
+                        t-att-class="'was-validated' if errors else None"
+                    >
+                        <input
+                            type="hidden"
+                            name="csrf_token"
+                            t-att-value="request.csrf_token()"
+                            t-nocache="The csrf token must always be up to date."
+                        />
+                        <div class="form-group">
+                            <label class="form-label" for="date_for_gift">
+                                Date for the gift
+                            </label>
+                            <input
+                                type="date"
+                                t-att-min="datetime.date.today()"
+                                required="required"
+                                class="form-control"
+                                id="date_for_gift"
+                                name="date_for_gift"
+                                t-att-value="date_for_gift"
+                            />
+                            <div
+                                t-if="errors.get('date_for_gift')"
+                                t-out="errors.get('date_for_gift')"
+                                class="invalid-feedback"
+                            />
+                        </div>
+                    </form>
+                </div>
+            </div>
+        </xpath>
+
+        <!-- Replace confirm button -->
+        <xpath expr="//a[@href='/shop/confirm_order']" position="attributes">
+            <attribute name="style">display: none;</attribute>
+        </xpath>
+
+        <xpath expr="//a[@href='/shop/confirm_order']" position="after">
+            <button id="gift_form_confirm_button" class="btn btn-primary mb32">
+                Confirm
+                <i class="fa fa-chevron-right" />
+            </button>
+        </xpath>
```

Comment:
hiding a button with css and adding another one is kind of ugly. can’t the original button be used differently instead (by overriding some maybe)?

Reply:
I’ve not yet find a better method for this particular purpose.

Reply:
why not use `position="replace"` then?